opal/datatype/opal_datatype_pack.c

```diff
@@ -410,14 +412,41 @@ pack_predefined_heterogeneous( opal_convertor_t* CONVERTOR,
                            ((ptrdiff_t)_r_blength == _elem->extent) ? "cont" : "----",
                            (void*)*(DESTINATION), (void*)_source, (unsigned long)_r_blength,
                            (unsigned long)(*(SPACE)) ); );
-    master->pFunctions[_elem->common.type]( CONVERTOR, _count,
-                                            _source, *SPACE, _elem->extent,
-                                            *DESTINATION, *SPACE, _r_blength,
-                                            &advance );
+// _count is the total number of basic datatypes in ELEM, not necessarily contiguous
```

Comment:
I would have gone the same way pack/unpack.h have been updated. There is a clear mechanism on how to deal with all the corner cases, and a nomenclature. This code uses a different one, making it more difficult to maintain in the long term.

Reply:
Are you talking about how I changed the buffer specifications from being "from" and "to" and switched to specifying them as "vector_buf" and "packed_buf" with a COPY_TO_VECTOR / COPY_FROM_VECTOR in "mode"?

The previous interface doesn't have enough arguments to describe the elem vector.  So if you're going to maintain the style as "from" and "to" then both of them have to become dt_elem_desc_t* because sometimes "from" is the vector and sometimes "to" is the vector. 

So you'd have pFunction take two dt_elem_desc_t* arguments for "from" and "to", but there would be an invisible requirement on the side that one of the two dt_elem_desc_t* descriptions had to be a degenerate case describing a packed buffer.  I could do that, but I'm not sure it's better